client/blocks/user-mention-suggestions/with-user-mention-suggestions.jsx

```diff
@@ -0,0 +1,34 @@
+/** @format */
+/**
+ * External Dependencies
+ */
+import React from 'react';
+
+/**
+ * withUserMentionSuggestions is a higher-order component that adds user mention support to whatever input it wraps.
+ *
+ * @example: withUserMentionSuggestions( Component )
+ *
+ * @param {object} EnhancedComponent - react component to wrap
+ * @returns {object} the enhanced component
+ */
+export default EnhancedComponent =>
+	class withUserMentionSuggestions extends React.Component {
+		static displayName = `withUserMentionSuggestions( ${ EnhancedComponent.displayName ||
+			EnhancedComponent.name } )`;
+		static propTypes = {};
+
+		handleKeyPress = e => {
+			if ( e.target.value[ e.target.value.length - 1 ] === '@' ) {
```

Comment:
I realize this is a placeholder, but I would suggest adding tests later to catch our desired cases. For example would we want `word@name` to trigger a highlight vs `word @bar`?